platform-sdk/swirlds-base/src/main/java/com/swirlds/base/internal/impl/BaseScheduledExecutorService.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.base.internal.impl;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * A scheduled executor service for the base modules. The executor is based on a single thread and is a daemon thread
+ * with a low priority.
+ * <p>
+ * This class is a singleton and should only be used by code in the base modules that highly needs an asynchronous
+ * executor.
+ */
+public class BaseScheduledExecutorService implements ScheduledExecutorService {
```

Comment:
Should we register a `ShutdownHook` to make sure we close this executor at some point?

Reply:
Good point